### PR TITLE
Update ray query as to framework

### LINKED ISSUE #367: Use the framework acceleration structures helper class in all ray tracing samples
@SaschaWillems and @JoseEmilio-ARM I updated the #313  PR to optionally use the framework helper class.  There's a few bugs in the framework so I'd like to let that PR get merged first and come back in a future PR to do the same treatment to this sample if that's okay with you guys?

_Originally posted by @gpx1000 in https://github.com/KhronosGroup/Vulkan-Samples/pull/317#discussion_r701615001_

### PR DESCRIPTION
**Pull request by gpx1000**
## Description

Migrate hand coded AS to Framework AS now that RayTracing Extended has been merged.

Fixes #367

## Tested Environments:
Ubuntu Linux
GeForce 2080
Vulkan version: 1.2.189.0

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [ ] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

**Review by JoseEmilio-ARM — APPROVED** [review]

**Review by SaschaWillems — APPROVED** [review]
LGTM

File: samples/extensions/ray_queries/ray_queries.h [hunk 1]
```diff
@@ -80,12 +74,12 @@ class RayQueries : public ApiVulkanSample
...
 	// Ray tracing structures
-	VkPhysicalDeviceAccelerationStructureFeaturesKHR acceleration_structure_features{};
-	AccelerationStructure                            top_level_acceleration_structure{};
-	AccelerationStructure                            bottom_level_acceleration_structure{};
-	uint64_t                                         get_buffer_device_address(VkBuffer buffer);
-	void                                             create_top_level_acceleration_structure();
-	void                                             create_bottom_level_acceleration_structure();
+	VkPhysicalDeviceAccelerationStructureFeaturesKHR  acceleration_structure_features{};
+	std::unique_ptr<vkb::core::AccelerationStructure> top_level_acceleration_structure    = nullptr;
+	std::unique_ptr<vkb::core::AccelerationStructure> bottom_level_acceleration_structure = nullptr;
```
Comment (mklefrancois) [inline samples/extensions/ray_queries/ray_queries.h:79]: I do not think the assignment to nullptr is needed here. I believe the default constructor of std::unique_ptr is doing it.

**Review by mklefrancois — APPROVED** [review]
Changes are LGTM, tested and runs on GeForce RTX 3060 Ti

**Comment by marty-johnson59** [comment]
Merging with 3 approvals.  LMK if issues

**Pull request merged**